### PR TITLE
Routing section

### PR DESCRIPTION
**Pull request by shannonbux**
Requesting feedback on: 
- position of this section
- docs within it
- names of the docs within it

### CODE REVIEW

**Comment by calcsam** [comment]
@amberleyromo I like that idea. The "cool" part about gatsby-link is the prefetching capabilities. So maybe we call it "Linking (and prefetching) with Gatsby" to highlight that. 

This is one of a category of "website features" that Gatsby gives you. I think we should:

* Rename "Adding third-party services" to something like "Adding typical website functionality"
* Add SEO, routing, accessibility, and i18n stub sections, alongside the current stub sections (search, analytics and forms). Authentication and payment could go in as well. At some point when the stories around personalization and A/B testing are more fleshed out, we can add those too.
* Move this routing section there.

**Comment by KyleAMathews** [comment]
>So maybe we call it "Linking (and prefetching) with Gatsby" to highlight that.

💯

**Comment by shannonbux** [comment]
Just updated this @calcsam @amberleyromo @KyleAMathews 

Will also rename and add redirects if we decide to go with these changes:

Adding third-party functionality --> Adding website functionality
gatsy-link--> Linking and prefetching with Gatsby

Looks like the section "Adding website functionality" is getting quite long. Not sure if it's too long?

IA is difficult!

**Comment by calcsam** [comment]
👍 looks good @shannonbux

File: www/src/data/sidebars/doc-links.yaml [hunk 1]
```diff
@@ -189,17 +185,23 @@
...
+        - title: Authentication*
+          link: /docs/authentication/
+        - title: E-commerce*
+          link: /docs/e-commerce/
+        - title: Interactive pages*
+          link: /docs/interactive-pages/
+        - title: Routing
+          link: /docs/routing/
+          items:
+          - title: @reach/router and Gatsby
```
Comment (DSchau) [inline www/src/data/sidebars/doc-links.yaml]: Heads up, the linter complained about this. Think you just need to wrap it in double quotes!

```
✖ bad indentation of a sequence entry at line 197, column 20:
              - title: @reach/router and Gatsby
```

**Comment by calcsam** [comment]
@amberleyromo I see your point -- why styling in the top bucket but routing in a sub-bucket? I don't really have a non-hand-wavy argument, but would argue that "styling" feels like a top-level category alongside "testing" and "debugging". I don't think this is blocking -- we can merge now and rearrange later if feels like routing belongs on the a higher level.

File: www/src/data/sidebars/doc-links.yaml (hunk 1 above)
Comment (calcsam) [reply www/src/data/sidebars/doc-links.yaml]: I've jumped in & added quotes to fix this.

**Pull request merged**